sys/src/9/port/devramfs.c

```diff
@@ -301,12 +307,12 @@ ramreadblock(Chan *c, void *buf, i32 n, i64 off)
 {
 	Proc *up = externup();
 	// first block, last block
-	int fb = off >> 13, fl = (off + n) >> 13, offinblock = off & 0x1fff;
+	int fb = off >> 21, fl = (off + n) >> 21, offinblock = off & 0x1fffff;
 
 	// adjust read size and offset.
 	// we only read one block for now.
 	if(fl != fb){
-		n = 8192 - (off & 0x1fff);
+		n = (1<<21) - (off & 0x1fffff);
 	}
 
 	if(fb > NBLOCK)
```

Comment:
Should these comparisons `fb > NBLOCK` be replaced with `fb > 16`?
Or better, replace NBLOCK with NPAGES, set to 16.